flang/include/flang/Lower/PFTBuilder.h

```diff
@@ -138,6 +138,11 @@ using Directives =
                parser::OpenACCDeclarativeConstruct, parser::OpenMPConstruct,
                parser::OpenMPDeclarativeConstruct, parser::OmpEndLoopDirective>;
 
+using Continues =
+    std::tuple<parser::ContinueStmt> ;
+template <typename A>
+static constexpr bool isContinueStmt{common::HasMember<A, Continues>};
+
```

Comment:
Kiran - `PFTBuilder.h` has a template for recognizing individual statements.  Given an `eval e`, it should be possible to query if `e.isA<parser::ContinueStmt>`.

However, if the idea is to find the block containing an "exit" `ContinueStmt`, it should be possible to get that automatically, in which case special case code should not be needed.  Like other constructs, a `DoConstruct` has a `constructExit` eval that should be set to that eval.  Depending on the `eval` passed to `createBodyOfOp` that `constructExit` eval may already be directly available.  Given that exit eval (and assuming it is the first eval in its block, which it should be), then its `block` member is the block of interest.

Reply:
Thanks, I completely missed that, I have switched to using isA<parser::ContinueStmt>.

Yes, good point about the constructExit. Question is what about for other non-looping constructs?

Also, going back to a point you raised in https://github.com/flang-compiler/f18-llvm-project/issues/1192, quoting you below.
"although there is some question as to whether the ContinueStmt should be inserted before or after the `OmpEndLoopDirective` Evaluation. (It may be that either way is ok.)"

I think the reason the branch might be missing is that the ContinueStmt is added after the `OmpEndLoopDirective` Evaluation. It might all work if it is before. I will give that a try.

Reply:
Yes, swapping the order of the `ContinueStmt` and the `OmpEndLoopDirective` is definitely worth a look.

Reply:
I am now using the `constructExit` to find the terminator block.

I tried to create the `ContinueStmt` before the `OmpEndLoopDirective` but that did not work. Not sure whether my approach for doing this was correct.